bundles/org.openhab.binding.evcc/src/main/java/org/openhab/binding/evcc/internal/EvccHandler.java

```diff
@@ -359,36 +363,22 @@ private void updateChannelsLoadpoint(int loadpointId) {
         channel = new ChannelUID(getThing().getUID(), loadpointName, CHANNEL_LOADPOINT_TARGET_SOC);
         updateState(channel, new QuantityType<>(targetSoC, Units.PERCENT));
         String targetTime = loadpoint.getTargetTime();
-        ZonedDateTime newTargetTimeZDT = ZonedDateTime.now().plusSeconds(30);
-        try {
-            Instant instant = Instant.parse(targetTime);
-            newTargetTimeZDT = ZonedDateTime.ofInstant(instant, ZoneId.systemDefault());
-        } catch (Exception e) {
+        if (targetTime == "null" || targetTime == null) {
+            channel = new ChannelUID(getThing().getUID(), loadpointName, CHANNEL_LOADPOINT_TARGET_TIME_ENABLED);
+            updateState(channel, OnOffType.OFF);
+            targetTimeEnabled = false;
+        } else {
             try {
-                newTargetTimeZDT = ZonedDateTime.parse(targetTime);
-            } catch (Exception f) {
-                logger.debug("Failed parsing targetTime {}. Error: {}.", targetTime, f.toString());
+                ZonedDateTime targetTimeZDT = ZonedDateTime.parse(targetTime);
+                channel = new ChannelUID(getThing().getUID(), loadpointName, CHANNEL_LOADPOINT_TARGET_TIME);
+                updateState(channel, new DateTimeType(targetTimeZDT));
+
+            } catch (NullPointerException e) {
```

Comment:
`NullPointerException` should not be caught, it should be avoided. What can cause it here?

Reply:
It can be caused when a String `null` is passed in to `ZonedDateTime.parse()`, I will avoid this by using `if (mysttring != null)`